src/main/java/seedu/address/logic/CommandArchive.java

```diff
@@ -0,0 +1,57 @@
+package seedu.address.logic;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.text.SimpleDateFormat;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Append all commands that user entered into command archive
+ */
+public class CommandArchive {
+    /**
+     * Create a logger without hard-coding the class name
+     */
+    private static final Logger LOGGER = Logger.getLogger(
+            Thread.currentThread().getStackTrace()[0].getClassName());
+
+    /**
+     * Takes in the userInputHistory string, extracts the latest command and writes it to commandFile.txt with a
+     */
+    public static void stringToFile (String inputString) {
+        File commandHistory = new File("commandFile.txt");
+        String latestUserCommand = inputString.substring(0, inputString.indexOf('\n'));
+        String timeStamp = new SimpleDateFormat("dd/MM/yyyy " + "HH:mm:ss  ").format(new java.util.Date());
+
+        try {
```

Comment:
Since this is in the Logic component, there should not be any writing to file in this class. That is the job of the Storage component. Having a clean separation of concerns will make it easier for debugging.